tests/transforms/test_diagonalize_measurements.py

```diff
@@ -128,7 +128,13 @@ def test_compound_observables(self, compound_obs, expected_res, base_obs):
 
     def test_legacy_hamiltonian(self):
         """Test that _diagonalize_observable works on legacy Hamiltonians observables"""
-        with pytest.warns():
+
+        if qml.operation.active_new_opmath():
+            with pytest.warns():
+                compound_obs = qml.ops.Hamiltonian([2, 3], [Y(0), X(1)])
+                expected_res = qml.ops.Hamiltonian([2, 3], [Z(0), Z(1)])
+                diagonalizing_gates, new_obs, visited_obs = _diagonalize_observable(compound_obs)
```

Comment:
If this test is only for the legacy Hamiltonian, then maybe we should just add the `legacy_opmath_only` fixture to this test.

Reply:
I believe we still wanted to make sure that the legacy Hamiltonian works with new opmath enabled?